tests/test_sims.py

```diff
@@ -882,3 +882,136 @@ def test_kwarg_passing(self):
+        A = mmsbm(self.n, self.p, self.alpha, rng=rng, directed=False, loops=False)
+
+        # check loopless and undirected
+        self.assertTrue(is_symmetric(A))
+        self.assertTrue(is_loopless(A))
+
+        # check dimensions
+        self.assertTrue(A.shape == (self.n, self.n))
+        pass
+
+    def test_bad_inputs(self):
+        with self.assertRaises(ValueError):
+            n = -1  # n must be greater than 0
+            mmsbm(n, self.p, self.alpha, rng=self.rng)
+
+        with self.assertRaises(TypeError):
+            n = "1"  # n must be an integer
+            mmsbm(n, self.p, self.alpha, rng=self.rng)
+
+        with self.assertRaises(TypeError):
+            n = 2.5  # n must be an integer
+            mmsbm(n, self.p, self.alpha, rng=self.rng)
+
+        with self.assertRaises(TypeError):
+            n = ["1", 10]  # n must be an integer
+            mmsbm(n, self.p, self.alpha, rng=self.rng)
+
+        with self.assertRaises(TypeError):
+            p = 0.5  # p must be an array or list
+            mmsbm(self.n, p, self.alpha, rng=self.rng)
+
+        with self.assertRaises(ValueError):
+            p = [[0.2, 0.1], ["str", 0.6]]  # p must only contain mumeric elements
+            mmsbm(self.n, p, self.alpha, rng=self.rng)
+
+        with self.assertRaises(ValueError):
+            p = [[5, 5], [4, 4]]  # p must be between 0 and 1
+            mmsbm(self.n, p, self.alpha, rng=self.rng)
+
+        with self.assertRaises(TypeError):
+            alpha = 1  # alpha must be an array
+            mmsbm(self.n, self.p, alpha, rng=self.rng)
+
+        with self.assertRaises(TypeError):
+            # test deafult alpha = None, should output typeerror
+            mmsbm(self.n, self.p, alpha, rng=self.rng)
+
+        with self.assertRaises(ValueError):
+            alpha = ["str", 0.3]  # alpha must only contain numeric elements
+            mmsbm(self.n, self.p, alpha, rng=self.rng)
+
+        with self.assertRaises(ValueError):
+            alpha = [0.1, -2]  # alpha entries must be positive
+            mmsbm(self.n, self.p, alpha, rng=self.rng)
+
+        with self.assertRaises(ValueError):
+            alpha = [
+                0.1,
+                0.5,
+                0.7,
+            ]  # dimension of p is 2x2 so alpha should be of length 2.
+            mmsbm(self.n, self.p, alpha, rng=self.rng)
+
+        with self.assertRaises(TypeError):
+            rng = 2  # must be generator
+            mmsbm(self.n, self.p, self.alpha, rng=rng)
+
+        with self.assertRaises(TypeError):
+            loops = 2  # loops must be a bool
+            mmsbm(self.n, self.p, self.alpha, loops=loops, rng=self.rng)
+
+        with self.assertRaises(TypeError):
+            directed = 2  # directed must be a bool
+            mmsbm(self.n, self.p, self.alpha, directed=directed, rng=self.rng)
+
+        with self.assertRaises(ValueError):
+            p = np.vstack(([0.6, 0.2], [0.3, 0.4]))  # directed must be a bool
```

Comment:
sorry what's being tested here?

Reply:
Apologies, the comment is not explanatory at all! 

This test should display an error because the graph is specified as undirected but a directed B matrix is provided.